src/test/daml/Daml/Finance/Instrument/Bond/Test/InterestRateSwap.daml

```diff
@@ -0,0 +1,102 @@
+  holdingFactoryCid <- toInterfaceContractId <$> submitMulti [custodian] [] do
+    createCmd Fungible.Factory with provider = custodian; observers = M.fromList pp
+
+  -- Create accounts
+  [custodianAccount, investorAccount] <- mapA (Account.createAccount "Default Account" [publicParty] accountFactoryCid holdingFactoryCid [] custodian) [custodian, investor]
+
+  -- Distribute commercial-bank cash
+  now <- getTime
+  let pp = [("PublicParty", singleton $ singleton publicParty)]
+  cashInstrumentCid <- Instrument.originate depository issuer "USD" "US Dollars" pp now
+
+  -- Create and distribute swap
+  -- Fix vs floating interest rate swap: Libor 3M vs 2.01% p.a. payment every 3M
+  let
+    issueDate = date 2019 Jan 16
+    firstPaymentDate = date 2019 Feb 15
+    maturityDate = date 2019 May 15
+    referenceRateId = "USD/LIBOR/3M"
+    issuerPaysFix = False
+    fixRate = 0.0201
+    paymentPeriod = M
+    paymentPeriodMultiplier = 3
+    dayCountConvention = Act360
+    businessDayConvention = ModifiedFollowing
+    firstPaymentAmount : Decimal = 1446.6167 -- The first payment is smaller than a regularly because the first period is shorter (only 1 month)
+    firstFixPaymentAmount : Decimal = 1675.0
+    firstFloatingPaymentAmount : Decimal = 228.3833
+    secondPaymentAmount : Decimal = 4466.0694 -- The second payment corresponds to a regular (3 month) period
+    secondFixPaymentAmount : Decimal = 4969.1667
+    secondFloatingPaymentAmount : Decimal = 503.0972
+    principalAmount = 1_000_000.0
+    observations = M.fromList [(dateToDateClockTime (date 2019 Jan 16), 0.0027406), (dateToDateClockTime (date 2019 Feb 15), 0.002035)]
+    holidayCalendarId = ["USD"]
+    cal =
+      HolidayCalendarData with
+        id = "USD"
+        weekend = [Saturday, Sunday]
+        holidays = [date 2019 Dec 19]
+
+  -- A reference data provider publishes the holiday calendar on the ledger
+  calendarCid <- submitMulti [calendarDataProvider] [] do
+    createCmd HolidayCalendar with
+      agency = calendarDataProvider
+      entity = cal.id
+      calendar = cal
+      observers = M.fromList pp
+
+  observableCid <- coerceContractId <$> submitMulti [issuer] [] do createCmd Observation with provider = issuer; obsKey = referenceRateId; observations; observers = M.empty
+
+  swapInstrument <- originateInterestRateSwap custodian issuer "SwapTest1" "Interest rate swap" pp now issueDate holidayCalendarId calendarDataProvider firstPaymentDate maturityDate dayCountConvention businessDayConvention fixRate paymentPeriod paymentPeriodMultiplier cashInstrumentCid referenceRateId issuerPaysFix
+  investorSwapTransferableCid <- Account.credit [publicParty] swapInstrument principalAmount investorAccount
+
+  -- One day before the first payment date: try to lifecycle and verify that there are no lifecycle effects.
+  verifyNoLifecycleEffects [publicParty] (subtractDays firstPaymentDate 1) swapInstrument settler issuer [observableCid]
```

Comment:
Again, use the `custodian` as the settler, and do away with the `settler` party. It simplifies understanding of the test.